Microsoft.Research/Contracts/MsCorlib/System.Collections.ObjectModel.Collection.cs

```diff
@@ -0,0 +1,394 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Reflection;
+using System.Runtime;
+using System.Runtime.InteropServices;
+
+namespace System.Collections.ObjectModel
+{
+    // Summary:
+    //     Provides the base class for a generic collection.
+    //
+    // Type parameters:
+    //   T:
+    //     The type of elements in the collection.
+    public class Collection<T> : IList<T>, ICollection<T>, IEnumerable<T>, IList, ICollection, IEnumerable
+    {
+        // Summary:
+        //     Initializes a new instance of the System.Collections.ObjectModel.Collection<T>
+        //     class that is empty.
+        public Collection()
+        {
+        }
+
+        //
+        // Summary:
+        //     Initializes a new instance of the System.Collections.ObjectModel.Collection<T>
+        //     class as a wrapper for the specified list.
+        //
+        // Parameters:
+        //   list:
+        //     The list that is wrapped by the new collection.
+        //
+        // Exceptions:
+        //   System.ArgumentNullException:
+        //     list is null.
+        public Collection(IList<T> list)
+        {
+            Contract.Requires(list != null);
+        }
+
+        // Summary:
+        //     Gets the number of elements actually contained in the System.Collections.ObjectModel.Collection<T>.
+        //
+        // Returns:
+        //     The number of elements actually contained in the System.Collections.ObjectModel.Collection<T>.
+        public int Count
+        {
+            get
+            {
+                return 0;
```

Comment:
Most of the stuff is auto-generated, and it would generate a huge effort to change everything, with low benefit.